src/BackgroundCells.jsx

```diff
@@ -33,20 +39,43 @@ class DisplayCells extends React.Component {
   }
 
   render(){
-    let { slots } = this.props;
+    let { slots, week, backgroundEvents, backgroundEventComponent,
+          backgroundEventPropGetter, ...props } = this.props;
     let { selecting, startIdx, endIdx } = this.state
 
     let children = [];
 
     for (var i = 0; i < slots; i++) {
+      let backgroundEvent = backgroundEvents.find(event => {
+        return dates.sameDay(event.start, week[i])
+      })
+
+      let Component = backgroundEventComponent;
+      let title = (backgroundEvent || {}).title;
+
+      var style, xClassName;
+      if (backgroundEvent && backgroundEventPropGetter)
+        ({ style, className: xClassName } = backgroundEventPropGetter(backgroundEvent));
+      else
+        ({ style, className: xClassName } = {});
+
       children.push(
         <div
           key={'bg_' + i}
-          style={segStyle(1, slots)}
-          className={cn('rbc-day-bg', {
+          style={{...segStyle(1, slots), ...style, ...props.style}}
+          className={cn('rbc-day-bg', xClassName, {
             'rbc-selected-cell': selecting && i >= startIdx && i <= endIdx
           })}
-        />
+        >
+        { Component && backgroundEvent
+          ? <Component
+              event={backgroundEvent}
+              eventPropGetter={backgroundEventPropGetter}
```

Comment:
this has already been used above, I wouldn't pass it down further if the prop has been consumed